### DISCUSSION TITLE
How To Upgrade Performance ?

### DISCUSSION THREAD

**Discussion by AFSDARIF** [filamentphp/filament#9520] [Help]
Currently my project has more than 50 tables, each table has more than 100 to 50.000 data or row. Incidentally my project is quite complex.

Before there was that much data, the performance was very fast, even when I tried to connect to a similar database but the data was small, the performance was still fast.

is it possible because now my database has thousands of data?

is it possible to optimize Eloquent with a Builder or other Query?

because, it seems that by default Filament carries out all the functions in the Model, maybe, I don't know.

**Comment by VincvanAs** [comment]
In order to receive suggestions, i would like to suggest you share some code of the queries you referring to

**Reply by AFSDARIF** [reply]
nothing different or special, I just followed what was in the documentation. https://filamentphp.com/docs/2.x/admin/resources/getting-started#creating-a-resource

50 more Models and each table has more than 100 to 50.000 data or row, make Filament very slowly. even when entering the login page

**Reply by sinnbeck** [reply]
Use either debugbar or clockwork to see what queries are run  on each page (as well as models created etc)

**Reply by VincvanAs** [reply]
Try php artisan icons:cache
